teamscale_client/client.py

```diff
@@ -121,22 +124,28 @@ def add_findings_group(self, name, mapping_pattern):
         Returns:
             requests.Response: request's response
         """
-        url = self.get_global_service_url('external-findings-group')
-        payload = [{'groupName': name, 'mapping': mapping_pattern}]
-        return self.put(url, payload)
+        url = "%s/%s" % (self.get_global_service_url('external-findings-group'), name)
+        return self.put(url, {'groupName': name, 'mapping': mapping_pattern})
 
     def add_finding_descriptions(self, descriptions):
         """Adds descriptions of findings.
 
         Args:
             descriptions (list): List of :class:`FindingDescription` to add to Teamscale.
         Returns:
-            requests.Response: request's response
+            str: 'success' if all descriptions were added.
         """
-
-        url = self.get_global_service_url('add-external-finding-descriptions')
-        payload = [{'typeId': d.typeid, 'description': d.description, 'enablement': d.enablement} for d in descriptions]
-        return self.put(url, payload)
+        base_url = self.get_global_service_url('external-findings-description')
+        for finding_description in descriptions:
+            some_description = dict()
+            some_description['typeId'] = finding_description.typeid
+            some_description['description'] = finding_description.description
+            some_description['enablement'] = finding_description.enablement
+            some_description['name'] = finding_description.name
+            url = "%s/%s" % (base_url, finding_description.typeid)
+            self.put(url, some_description)
+
+        return 'success'
```

Comment:
I think we should save at least one response from the server and return the worst one. i.e. if an error occurs, return that, if everything is fine just return the last success. Right now this changes the API for the user, which is suboptimal.